java/code/src/com/redhat/rhn/taskomatic/task/ReportDBHelper.java

```diff
@@ -48,6 +49,17 @@ public static <T> Stream<DataResult<T>> batchStream(SelectMode query, int batchS
                 .takeWhile(batch -> !batch.isEmpty());
     }
 
+    /**
+     * Generated a query for checking if a table exists
+     * @param session session the query should use
+     * @param table table name
+     * @return select mode query
+     */
+    public static SelectMode generateExistsTable(Session session, String table) {
+        final String sqlStatement = "SELECT to_regclass('" + table + "') IS NOT NULL;";
```

Comment:
I would use an alias here in the query so you can refer to it in `checkRemoteData()` instead of using `?column?`

Reply:
```suggestion
        final String sqlStatement = "SELECT to_regclass('" + table + "') IS NOT NULL AS  table_exists;";
```

What I meant was to define the alias in the query with `AS`, so that the result column has a readable name that you can use as `COLUMN_KEY`.